### ISSUE TITLE
Extension doesn't seem to respect acceptSelectedSuggestion

### ISSUE THREAD

**Issue by figurae** [mborik/z80-macroasm-vscode#30]
Hi! Thanks for this great extension. I seem to have a problem though.

It seems that code completion for things like register and label names provided by this extension is handled differently from other completions. When I change the keybinding for `acceptSelectedSuggestion` from `Enter` to something else, pressing enter still completes the selection, except now it also moves the cursor down. Snippets respect the change, though.

**Comment by mborik** [comment]
Hi. Thank you for reporting, but I honestly can't imagine your situation and the problem that is happening to you at all. Can you give me more details, your extension settings and specific VSCode settings? A test scenario that I could simulate and reproduce would help (or short video).

**Comment by figurae** [comment]
Certainly. I'm doing this on a fresh VSCode profile and the only extension installed is Z80 Macro-Assembler. All settings are left at default values.

Now, if I open a `.js` file and type `cons`, I get a code completion popup:

![001](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/6e4ce173-3c50-44bc-a31d-fc329f979623)

When I press enter, the code completes with the cursor remaining at the end of the same line:

![002](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/3add9541-6ea4-4166-928c-231480835e75)

Now, let's move to an `.asm` file, which triggers the Z80 Macro-Assembler extension. I type `[tab]ld h` to get the code completion popup:

![003](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/6161bfab-519b-48cd-babc-79da5856b160)

Now, when I press enter on `hl`, I get the same behaviour as in the `.js` file, `hl` is completed and the cursor remains at the end of the same line:

![004,png](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/24ab388b-4f77-4838-8f53-ac5dd3d435c9)

On a side note, I noticed that this only works in this manner for the first register, if I try to complete the second register, like so:

![005](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/a289648b-4e25-48d9-b0de-7ca8abf57f38)

Pressing enter not only completes the register name, but also moves the cursor down to the next line:

![006](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/17bcc3bc-3254-4953-b7ce-9e9522f439b8)

I understand the reasoning behind this, but I think it would be nice if this behaviour was adjustable, as I often like to write a comment on the same line after the second register. Anyway, back to the issue at hand. ;)

Now, let's change the default keybinding for accepting popup selections, i.e., `acceptSelectedSuggestion`:

![007](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/f9fb47de-c6f7-454d-8376-35b262acc3c1)

Normally, I like to use something like `ctrl+enter`, but let's change it to `f` to make certain there's no `enter` anywhere:

![008](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/fdfd07fe-d3d8-43eb-86bd-b679c1d7b56b)

Now, when I type `cons` in the `.js` file, see the completion popup, and press enter, I get:

![009](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/95901c13-80a4-412a-aa33-f13a26fd41b2)

Which is the expected result as `enter` is no longer bound to the completion action. However, when we move to the `.asm` file and attempt to press enter after `[tab]ld h` and moving the cursor to `hl`, we get:

![010](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/76ea1710-4ed8-4906-a799-122667d2436e)

So the code still completes with `enter`, even though the key is no longer bound to this action. Additionally, the cursor moves to the next line, as, I think, it both accepts the completion and reads `enter` as a normal press of an enter key. When I press `enter` after `[tab]ld hl,b` and selecting `bc` from the list, I get this:

![011](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/cc26f657-dcd6-4426-a2e4-23332f7f735c)

The cursor is now two lines down. I think it's because the code completion action includes one enter when completing the second register and now it also registers the keypress as a normal enter as well, resulting in two enters.

Expected behaviour: I would expect the code not to complete at all when pressing a key that is not bound to the `acceptSelectedSuggestion` action. In such case, I would expect the editor to just react to the key like normal, ignoring code completion, like this:

![012](https://github.com/mborik/z80-macroasm-vscode/assets/89541173/e3f4f0a3-00ef-436a-a3c8-1ef551d732ef)

And I would also expect the code completion action not to move the cursor down depending on whether I'm completing the first or the second register name (does this need a separate issue, or is it okay if I just write it here?)

**Comment by mborik** [comment]
@figurae Well, I wasn't able to find a better fix than that I added a new configuration property `suggestCommitWithCustomKeys` which is disabled by default, so, it will not modify your own commit characters for suggestions (defined by `acceptSelectedSuggestion` shortcut).
It working better this way in general. 🙂
